File: client/src/api/import_.ts
import apiClient from './client'

export interface ColumnMappingItem {
  csv_column: string
  field: string | null
}

export interface ImportPreviewResponse {
  rows: Record<string, string>[]
  columns: string[]
  suggested_mapping: ColumnMappingItem[]
  total_rows: number
}

export interface ImportConfirmResponse {
  imported: number
  duplicates: number
  invalid: number
  total: number
  errors: string[]
}

export const FIELD_OPTIONS = [
  { value: 'email',       label: 'Email (required)' },
  { value: 'nickname',    label: 'Nickname' },
  { value: 'platform',    label: 'Platform' },
  { value: 'followers',   label: 'Followers' },
  { value: 'profile_url', label: 'Profile URL' },
  { value: 'industry',    label: 'Industry' },
]

export const importApi = {
  preview: (file: File): Promise<ImportPreviewResponse> => {
    const form = new FormData()
    form.append('file', file)
    return apiClient.post<ImportPreviewResponse>('/influencers/import/preview', form, {
      headers: { 'Content-Type': 'multipart/form-data' },
    }).then(r => r.data)
  },

  confirm: (
    file: File,
    mapping: ColumnMappingItem[],
    overwrite: boolean,
  ): Promise<ImportConfirmResponse> => {
    const form = new FormData()
    form.append('file', file)
    form.append('mapping', JSON.stringify(mapping))
    form.append('overwrite_duplicates', String(overwrite))
    return apiClient.post<ImportConfirmResponse>('/influencers/import/confirm', form, {
      headers: { 'Content-Type': 'multipart/form-data' },
    }).then(r => r.data)
  },
}

File: server/app/schemas/import_.py
from typing import Any
from pydantic import BaseModel


class ColumnMappingItem(BaseModel):
    csv_column: str        # column name as it appears in the file
    field: str | None      # target field name, None = skip


class ImportPreviewResponse(BaseModel):
    rows: list[dict[str, Any]]          # first 10 rows (raw values)
    columns: list[str]                  # all column names from file
    suggested_mapping: list[ColumnMappingItem]
    total_rows: int                     # total rows excluding header


class ImportRowResult(BaseModel):
    row_index: int
    email: str
    status: str        # "imported" | "duplicate" | "invalid_email" | "skipped"
    message: str | None = None


class ImportConfirmResponse(BaseModel):
    imported: int
    duplicates: int
    invalid: int
    total: int
    errors: list[str]
